dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/util/json/DatasetVersionDTOTest.java

```diff
@@ -51,50 +50,43 @@ public void setUp() {
+        author1Fields.add(FieldDTO.createVocabFieldDTO("authorIdentifierScheme", "ORCID"));
+        author1Fields.add(FieldDTO.createPrimitiveFieldDTO("authorName", "Privileged, Pete"));
 
-            author1Fields.add(FieldDTO.createPrimitiveFieldDTO("authorAffiliation", "Top"));
-            author1Fields.add(FieldDTO.createPrimitiveFieldDTO("authorIdentifier", "ellenid"));
-            author1Fields.add(FieldDTO.createVocabFieldDTO("authorIdentifierScheme", "ORCID"));
-            author1Fields.add(FieldDTO.createPrimitiveFieldDTO("authorName", "Privileged, Pete"));
+        HashSet<FieldDTO> author2Fields = new HashSet<>();
 
-            HashSet<FieldDTO> author2Fields = new HashSet<>();
+        author2Fields.add(FieldDTO.createPrimitiveFieldDTO("authorAffiliation", "Bottom"));
+        author2Fields.add(FieldDTO.createPrimitiveFieldDTO("authorIdentifier", "audreyId"));
+        author2Fields.add(FieldDTO.createVocabFieldDTO("authorIdentifierScheme", "DAISY"));
+        author2Fields.add(FieldDTO.createPrimitiveFieldDTO("authorName", "Awesome, Audrey"));
 
-            author2Fields.add(FieldDTO.createPrimitiveFieldDTO("authorAffiliation", "Bottom"));
-            author2Fields.add(FieldDTO.createPrimitiveFieldDTO("authorIdentifier", "audreyId"));
-            author2Fields.add(FieldDTO.createVocabFieldDTO("authorIdentifierScheme", "DAISY"));
-            author2Fields.add(FieldDTO.createPrimitiveFieldDTO("authorName", "Awesome, Audrey"));
+        List<HashSet<FieldDTO>> authorList = new ArrayList<>();
+        authorList.add(author1Fields);
+        authorList.add(author2Fields);
+        FieldDTO expectedDTO = new FieldDTO();
+        expectedDTO.setTypeName("author");
+        expectedDTO.setMultipleCompound(authorList);
 
-            List<HashSet<FieldDTO>> authorList = new ArrayList<>();
-            authorList.add(author1Fields);
-            authorList.add(author2Fields);
-            FieldDTO expectedDTO = new FieldDTO();
-            expectedDTO.setTypeName("author");
-            expectedDTO.setMultipleCompound(authorList);
+        FieldDTO authorDTO = dto.getMetadataBlocks().get("citation").getFields().get(1);
 
-            FieldDTO authorDTO = dto.getMetadataBlocks().get("citation").getFields().get(1);
+        // write both dto's to json to compare them with gson parser
+        JsonElement expected = gson.toJsonTree(expectedDTO, FieldDTO.class);
+        JsonElement result = gson.toJsonTree(authorDTO);
 
-            // write both dto's to json to compare them with gson parser
-
-            JsonElement expected = gson.toJsonTree(expectedDTO, FieldDTO.class);
-            JsonElement result = gson.toJsonTree(authorDTO);
-
-            Assert.assertEquals(expected, result);
-
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
+        assertEquals(expected, result);
     }
 
-
+    private String readFileToString(String resourcePath) throws IOException {
```

Comment:
I copied the function here from `ExportServiceTest` and I don't like this code duplication.

Is there a class for helper functions where I could move `readFileToString` to?

Reply:
Valid point, atm we don't have any util test classes. At the other hand this is just an delegation to library code so using this line in the main test method wouldn't be "wrong" and then you can't really call it duplicate. So I'm not sure if it is worth to extract to another class. But I'm kinda 50/50 on that.

Reply:
Your call then - if it's OK for, I'm fine with merging the code as it it.

I could add a TODO or FIXME note, but I'm not sure if its value will outweigh the clutter.

Reply:
I think I'm fine with it, I wouldn't add any TODO, just check for yourself how many todo's are in the project, and then check the date's near todo using git annotate ;).